### PR TITLE
Push buildpack sorting logic to repo layer

### PR DESCRIPTION
**Pull request by gogolok**
## Is there a related GitHub Issue?
 #3541

## What is this change about?
Moving the sorting logic for 'buildpacks' to the repository layer.

## Does this PR introduce a breaking change?
No.

### CODE REVIEW

File: api/repositories/buildpack_repository_test.go [hunk 1]
```diff
@@ -34,7 +51,7 @@ var _ = Describe("BuildpackRepository", func() {
 			})
 
 			It("returns all buildpacks", func() {
```
[the remark this answers is not in the document]
Comment (gogolok) [reply api/repositories/buildpack_repository_test.go]: I added some additional code to verify that the sorter has been invoked.

File: api/repositories/buildpack_repository.go [hunk 2]
```diff
@@ -34,15 +37,64 @@ type BuildpackRecord struct {
...
+		sorter: compare.NewSorter(BuildpackComparator),
+	}
+}
+
+func (s *buildpackSorter) Sort(records []BuildpackRecord, order string) []BuildpackRecord {
+	return s.sorter.Sort(records, order)
+}
+
+func BuildpackComparator(fieldName string) func(BuildpackRecord, BuildpackRecord) int {
+	return func(d1, d2 BuildpackRecord) int {
```
[the remark this answers is not in the document]
Comment (gogolok) [reply api/repositories/buildpack_repository.go]: Good catch :-) I've changed it.

File: api/repositories/buildpack_repository_test.go [hunk 3]
```diff
@@ -57,11 +80,34 @@ var _ = Describe("BuildpackRepository", func() {
 					}),
 				))
 			})
+
+			Describe("ordering", func() {
```
Comment (danail-branekov) [inline api/repositories/buildpack_repository_test.go:84]: In order to improve test readability and reduce the nesting level, how about setting the `OrderBy` field on the message in the `Describe("list"` 's `BeforeEach` and have the `It("sorts the buildpacks` on the same level as `It("returns all buildpacks"`?

**Review by danail-branekov — CHANGES_REQUESTED** [review]

File: api/repositories/buildpack_repository_test.go (hunk 3 above)
Comment (gogolok) [reply api/repositories/buildpack_repository_test.go]: I've tried to follow your suggestion. Is it now what you've expected?

**Review by danail-branekov — APPROVED** [review]

**Pull request merged**